packages/augur-core/src/contracts/trading/erc20proxy1155/ERC20Proxy1155Nexus.sol

```diff
@@ -0,0 +1,84 @@
+pragma solidity 0.5.15;
+
+import 'ROOT/libraries/CloneFactory.sol';
+import 'ROOT/libraries/token/IERC20.sol';
+import 'ROOT/libraries/token/IERC1155.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+
+import 'ROOT/trading/erc20proxy1155/IERC20Proxy1155Nexus.sol';
+import 'ROOT/trading/erc20proxy1155/IERC20Proxy1155.sol';
+
+// To use these ERC20s, the user will have to call
+// setApprovalForAll(ERC20Proxy1155Nexus, true)
+// on the ShareToken contract themselves.
+
+contract ERC20Proxy1155Nexus is IERC20Proxy1155Nexus {
+    using SafeMathUint256 for uint256;
+
+    IERC1155 public target1155;
+    IERC20Proxy1155 proxyToClone;
+
+    // tokenId -> owner -> spender -> allowance
+    // The spender is whoever called the proxy contract, which is msg.sender in the proxy and _caller here in the nexus.
+    mapping (uint256 => mapping (address => mapping (address => uint256))) public allowances;
+    // tokenId -> proxy contract
+    mapping (uint256 => IERC20Proxy1155) public proxies;
+
+    constructor(address _proxyToClone, IERC1155 _target1155) public {
+        proxyToClone = IERC20Proxy1155(_proxyToClone);
+        target1155 = _target1155;
+    }
+
+    function newERC20(uint256 _tokenId) public returns (IERC20Proxy1155) {
```

Comment:
Would be nice to have a convenience function to create all the ERC20s for a given market too. I suspect that's what will normally be used since just one cant really be used effectively

Reply:
That's true. I'll make a bulk version.